# Sequencing Reads:
We use the following three different real sequencing data:

## PacBio HiFi reads:
```
https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/data/AshkenazimTrio/HG003_NA24149_father/PacBio_CCS_15kb_20kb_chemistry2/reads/PBmixSequel729_1_A01_PBTH_30hours_19kbV2PD_70pM_HumanHG003.fastq.gz
```


## ONT ultra-long reads:
```
https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/data/AshkenazimTrio/HG003_NA24149_father/UCSC_Ultralong_OxfordNanopore_Promethion/GM24149_1.fastq.gz
```

## Illumina 101bp reads:
```
https://www.ebi.ac.uk/ena/browser/view/ERR194147
```
---

# CAMI Metagenomic Reads:
https://data.cami-challenge.org/
## CAMI Low Complexity
```
RL_S001__insert_270.fq
```
## CAMI High Complexity
```
RH_S001__insert_270.fq
```

---

# Reference Genomes:

## The complete Human Genome GRCh38.p14 (GCF_000001405.40, release date 3 February 2022) 

```
https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.40
```

## The human genome chromosome 1 (GCF_000001405.40, release date 3 February 2022)

```
https://www.ncbi.nlm.nih.gov/nuccore/NC_000001.11
```

## The largest sequenced reference genome, pinus taeda (also known as loblolly pine, GCA_000404065.3, release date 9 January 2017)

```
https://www.ncbi.nlm.nih.gov/assembly/GCA_000404065.3/
```

---
 
# Metagenomes:
We obtain our metagenomes for building the reference database from RefSeq (https://www.ncbi.nlm.nih.gov/refseq) database.
The list of TAXIDs for the metagenomes we choose is listed as follows:
## RefSeq1
```
RefSeq1_taxids.txt
```
## RefSeq2
```
RefSeq2_taxids.txt
```


